### ISSUE TITLE
'Must have completion callback with nowait=False' if closing channel with existing consumers

### ISSUE THREAD

**Issue by desertkun** [pika/pika#817]
If you have existing consumers and decide to close a channel, you will end up with an error:
```python
ValueError: Must have completion callback with nowait=False
```
That's because [this line](https://github.com/pika/pika/blob/master/pika/channel.py#L529) definitely doing something wrong (it passes NO callback and sets nowait=False)

**Comment by lukebakken** [comment]
Resolved by #818, thanks!

**Comment by behconsci** [comment]
I am still getting this error with version 0.11.2

```File "/usr/local/lib/python3.6/site-packages/pika/channel.py", line 1125, in _on_deliver
    header_frame.properties, body)
  File "/usr/local/lib/python3.6/site-packages/asgi_rabbitmq/core.py", line 195, in consume_message
    amqp_channel.basic_cancel(consumer_tag=tag)
  File "/usr/local/lib/python3.6/site-packages/pika/channel.py", line 231, in basic_cancel
    'Must have completion callback with nowait=False')
ValueError: Must have completion callback with nowait=False

    amqp_channel.basic_cancel(consumer_tag=tag)
  File "/usr/local/lib/python3.6/site-packages/pika/channel.py", line 231, in basic_cancel
    'Must have completion callback with nowait=False')
ValueError: Must have completion callback with nowait=False```

**Comment by lukebakken** [comment]
@behconsci please provide code to reproduce. Pika's maintainers do so on what amounts to their free time, so adding time to reproduce an issue when only a stack trace is provided hinders fixing it.

**Comment by lukebakken** [comment]
@behconsci - here's the problem. The [default parameters](https://github.com/pika/pika/blob/master/pika/channel.py#L202) for `basic_cancel` *will* cause that exception to happen every time, because `nowait` defaults to `False` which requires a callback. `asgi_rabbitmq` has already addressed this [here](https://github.com/proofit404/asgi_rabbitmq/blob/master/asgi_rabbitmq/core.py#L201-L202), so you need to upgrade `asgi_rabbitmq`.

We may consider changing the method signature to prevent this from happening in the future.

**Comment by behconsci** [comment]
thanks @lukebakken! I upgraded `asgi_rabbitmq` and it worked again.

**Comment by lukebakken** [comment]
@behconsci thank you for letting me know that fixed your issue.

**Comment by vitaly-krugl** [comment]
@lukebakken, this should be fixed now because we removed the `nowait` parameter, right?

**Comment by lukebakken** [comment]
Yep, resolved by #941
